### PR TITLE
[New post] Java REST API with Spring Boot

### PR DESCRIPTION
**Pull request by mraible**
This blog post has the following features:

- [x] A GitHub Repository with a polished README
- [x] A GitHub Repository under the github.com/oktadev account
- [x] A title that's approved by Dev Advocacy
- [x] A URL that's approved by Dev Advocacy
- [x] The content has been run through Grammarly (https://www.grammarly.com/)
- [x] Rendered locally and confirmed that no Markdown typos exist

### CODE REVIEW

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+toc::[]
+
+== Install a JDK with GraalVM
+
+Use SDKMAN to install Java 17 with GraalVM
+
+  sdk install java 22.1.0.r17-grl
+
+== Generate an OAuth 2.0 Access Token
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
== Generate an OAuth 2.0 access token
```

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+. Install the https://cli.okta.com/[Okta CLI] and run `okta register` to sign up for a new account. If you already have an account, run `okta login`.
+
+. Run `okta apps create spa`. Set `oidcdebugger` as an app name and press **Enter**.
+
+. Use `\https://oidcdebugger.com/debug` for the Redirect URI and set the Logout Redirect URI to `\https://oidcdebugger.com`.
+
+. Navigate to the https://oidcdebugger.com/[OpenID Connect Debugger website].
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
. Navigate to the https://oidcdebugger.com/[OpenID Connect debugger website].
```
matches the site capitalization

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .authorizeRequests(request -> request.anyRequest().authenticated())
+            .oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
+    }
+}
+----
+
+=== Run and Test Your Spring Boot REST API with HTTPie
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
=== Run and test your Spring Boot REST API with HTTPie
```

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+. Start your app from your IDE or using a terminal:
+
+  ./mvnw spring-boot:run
+
+. Test your API with an access token.
+
+  http :8080/hello Authorization:"Bearer $TOKEN"
+
+=== Build a Native Spring Boot App
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
=== Build a native Spring Boot App
```

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+. Start your Spring Boot app:
+
+  ./target/demo
+
+. Test your API with an access token.
+
+  http :8080/hello Authorization:"Bearer $TOKEN"
+
+== Startup Time Comparison
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
== Startup time comparison
```

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+
+. Run each image three times before recording the numbers, then each command five times
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
. Run each image three times before recording the numbers, then run each command five times.
```
punctuation. clarity.
Comment: This step is numbered "1." but I don't actually see a step 2.  Would it make sense to remove the numbering?

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+      title: 'Java Framework'
+    }
+  };
+  var chart = new google.visualization.BarChart(document.getElementById('startup-times'));
+  chart.draw(data, options);
+}
+</script>
+++++
+
+== Memory Usage Comparison
```
Comment (havi) [inline _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]:
```suggestion
== Memory usage comparison
```

**Review by havi — CHANGES_REQUESTED** [review]
Edits: Sentence case for section headings and one comment about numbering.

File: _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc (hunk 6 above)
Comment (mraible) [reply _source/_posts/2022-05-26-java-rest-api-spring-boot.adoc]: Yes. I'll make it match the one we published today.

**Review by havi — APPROVED** [review]
Ready to ship!

**Pull request merged**